workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/RuntimeConfig.kt

```diff
@@ -20,19 +20,44 @@ public annotation class WorkflowExperimentalRuntime
  */
 public sealed interface RuntimeConfig {
   /**
-   * This version of the runtime will process as many actions as possible after one is received
-   * until [frameTimeoutMs] has passed, at which point it will render().
+   * This version of the runtime will process all actions queued within [actionWaitMs]
+   * (i.e. cascade up the tree only) before calling render() once. If [frameTimeoutMs] has passed
+   * while processing actions, it will stop and call render(). After render() is called it will
+   * pass the new rendering to the UI (in other words it can't process forever if actions are
+   * queueing faster than [actionWaitMs]).
+   *
+   * i.e. this is one "Render Pass" per 'frame'.
    */
   @WorkflowExperimentalRuntime
-  public data class FrameTimeout(public val frameTimeoutMs: Long = 30L) : RuntimeConfig
+  public data class RenderPassPerFrame(
+    public val actionWaitMs: Long = 10L,
+    public val frameTimeoutMs: Long = 300L
+  ) : RuntimeConfig
+
+  /**
+   * This version of the runtime will process all actions queued within [actionWaitMs] AND call
+   * render() after processing them (i.e. cascade up and down the tree) before passing the
+   * updated rendering to the UI or until [frameTimeoutMs] has passed, at which point it will
+   * pass the new rendering to the UI (in other words it can't process forever if actions are
+   * queueing faster than [actionWaitMs]).
+   *
+   * i.e. this is one "Rendering" per 'frame'.
+   */
+  @WorkflowExperimentalRuntime
+  public data class RenderingPerFrame(
```

Comment:
Sorry to bike shed, but I'm having a hard time distinguishing btw. the two names. Perhaps:

- `ExhaustActionsBeforeRenderPass`
- `DropExtraRenderings`

Reply:
Oh no, I thought these names were so smart. Maybe we should have a quick huddle to decide so it minimizes the thrash.